scripts/morty/isobar_morty_pde.py

```diff
@@ -0,0 +1,502 @@
+        solving each individually in a Jacobi-like manner.
+        This approach will provide a more accurate result,
+        as contributions from other nuclides in the isobar
+        will be included.
+        The next step up from this approach is to incorporate
+        the spatial solve within the depletion solver itself.
+
+        Parameters
+        ----------
+        nodes : int
+            Number of spatial nodes
+        z1 : float
+            Position of in-core to ex-core transition
+        z2 : float
+            Position of ex-core to in-core transition
+        nu1 : float
+            Velocity in-core
+        nu2 : float
+            Velocity ex-core
+        lmbda : float
+            Time step times velocity divized by spatial mesh size
+        tf : float
+            Final time
+        lams : dict
+            key : str
+                Isobar nuclide indicator (a, b, c, d, or d_m1)
+            value : float
+                Decay constant for given nuclide
+        FYs : dict
+            key : string
+                Isobar nuclide indicator (a, b, c, d, or d_m1)
+            value : float
+                Fission yield for nuclice
+        br_c_d : float
+            Branching ratio for nuclide "c" to "d"
+        br_dm1_d : float
+            Branching ratio for nuclide "dm1" to "d"
+        vol1 : float
+            Volume of in-core region
+        vol2 : float
+            Volume of ex-core region
+        losses : dict
+            key : string
+                Isobar nuclide indicator (a, b, c, d, or d_m1)
+            value : float
+                Loss term due to parasitic absorption or other similar terms
+
+        """
+        self.nodes = nodes
+        self.z1 = z1
+        self.zs = np.linspace(0, z1+z2, nodes)
+        self.dt = lmbda * dz / nu1
+        self.ts = np.arange(0, tf+self.dt, self.dt)
+        self.nu_vec = self._format_spatial(nu1, nu2)
+
+        self.mu = {}
+        self.mu['a'] = self._format_spatial((lams['a'] + losses['1a']),
+                                            (lams['a'] + losses['2a']))
+        self.mu['b'] = self._format_spatial((lams['b'] + losses['1b']),
+                                            (lams['b'] + losses['2b']))
+        self.mu['c'] = self._format_spatial((lams['c'] + losses['1c']),
+                                            (lams['c'] + losses['2c']))
+        self.mu['d_m1'] = self._format_spatial((lams['d_m1']
+                                                + losses['1d_m1']),
+                                               (lams['d_m1'] + losses['2d']))
+        self.mu['d'] = self._format_spatial((lams['d'] + losses['1d']),
+                                            (lams['d'] + losses['2d']))
+        self.S = {}
+        self.S['a'] = self._format_spatial((FYs['a']/vol1), (0/vol2))
+
+        self.lama = lams['a']
+        self.lamb = lams['b']
+        self.lamc = lams['c']
+        self.lamd = lams['d']
```

Comment:
why is this hardcoded? What is the point of having dictionary accessors if we simply assign variables for every value? i don't understand.